src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,185 @@
+
+        private:
+            std::string instring; //todo change to IReferenceCounted file
+        };
+    };
+
+    struct custom_preprocessing_hooks : public boost::wave::context_policies::default_preprocessing_hooks{
+
+        custom_preprocessing_hooks(const nbl::asset::IShaderCompiler::SPreprocessorOptions& _preprocessOptions, std::vector<std::pair<uint32_t, std::string>>&& _lineOffsetStack) : preprocessOptions(_preprocessOptions), lineOffsetStack(_lineOffsetStack) {}
+
+        nbl::asset::IShaderCompiler::SPreprocessorOptions preprocessOptions;
+        std::vector<std::pair<uint32_t, std::string>> lineOffsetStack;  //TODO no longer needed
+
+
+        template <typename ContextT>
+        bool
+            locate_include_file(ContextT& ctx, std::string& file_path,
+                bool is_system, char const* current_name, std::string& dir_path,
+                std::string& native_name) {
+
+            //on builtin return true
+            //default returns false if file does not exist
+            if (nbl::builtin::hasPathPrefix(file_path))
+            {
+                file_path = file_path.substr(nbl::builtin::pathPrefix.size() + 1);
+                return true;
+            }
+            if (!ctx.find_include_file(file_path, dir_path, is_system, current_name))
+                return false;   // could not locate file
+
+            namespace fs = boost::filesystem;
+
+            fs::path native_path(boost::wave::util::create_path(file_path));
+            if (!fs::exists(native_path)) {
+                //TODO throw error or log message?
+                /*BOOST_WAVE_THROW_CTX(ctx, boost::wave::preprocess_exception, boost::wave::bad_include_file,
+                    file_path.c_str(), ctx.get_main_pos());*/
+                //preprocessOptions.logger.log("Pre-processor error ...:\n%s", nbl::system::ILogger::ELL_ERROR, message);
+
+                return false;
+            }
+            return true;
+        }
+
+        //template <typename ContextT>
+        //void opened_include_file(ContextT const& ctx, std::string const& relname, std::string const& absname, bool is_system_include) {
+        //    //if (preprocessOptions.includeFinder)
+        //    //{
+        //    //    return getInputStreamInclude(
+        //    //        preprocessOptions.includeFinder, m_system.get(), preprocessOptions.maxSelfInclusionCount + 1u,
+        //    //        preprocessOptions.sourceIdentifier.data(), path.c_str(), !isSystemPath,
+        //    //        lexer.GetCurrLineIndex(), leadingLinesImports, lineOffsetStack
+        //    //    );
+        //    //}
+        //    //else
+        //    //{
+        //    //    lineOffsetStack.push_back(lineOffsetStack.back()); // have to push something because pop-include expects something
+        //    //    return tcpp::TInputStreamUniquePtr(new tcpp::StringInputStream(std::string("#error No include handler\n")));
+        //    //}
+        //    _NBL_DEBUG_BREAK_IF(true);
```

Comment:
the only thing you need to do here is just insert `#line 1 "absname"`, and nothing else

Reply:
ah nothing needed